ngraph/test/frontend/paddlepaddle/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "paddlepaddle_tests")
+
+file(GLOB_RECURSE SRC ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+
+add_executable(${TARGET_NAME} ${SRC})
+
+target_link_libraries(${TARGET_NAME} PRIVATE frontend_shared_test_classes)
+
+add_clang_format_target(${TARGET_NAME}_clang FOR_TARGETS ${TARGET_NAME})
+
+install(TARGETS ${TARGET_NAME}
+        RUNTIME DESTINATION tests
+        COMPONENT tests
+        EXCLUDE_FROM_ALL)
+
+# Test model generating
+ie_check_pip_package(paddlepaddle WARNING)
+
+set(TEST_PADDLE_MODELS_DIRNAME test_model_zoo/paddle_test_models)
+target_compile_definitions(${TARGET_NAME} PRIVATE -D TEST_PADDLE_MODELS_DIRNAME=\"${TEST_PADDLE_MODELS_DIRNAME}/\")
+
+# If 'paddlepaddle' is not found, code will still be compiled
+# but models will not be generated and tests will fail
+# This is done this way for 'code style' and check cases - cmake shall pass, but CI machine doesn't need to have
+# 'paddlepaddle' installed to check code style
+if (paddlepaddle_FOUND)
+    set(TEST_PADDLE_MODELS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${TEST_PADDLE_MODELS_DIRNAME}/)
+
+    file(GLOB_RECURSE PADDLE_GEN_SCRIPTS ${CMAKE_CURRENT_SOURCE_DIR}/test_models/gen_scripts/generate_*.py)
+    file(GLOB_RECURSE PADDLE_ALL_SCRIPTS ${CMAKE_CURRENT_SOURCE_DIR}/*.py)
+    set(OUT_FILES "")
+    foreach(GEN_SCRIPT ${PADDLE_GEN_SCRIPTS})
+        get_filename_component(FILE_WE ${GEN_SCRIPT} NAME_WE)
+        set(OUT_DONE_FILE ${TEST_PADDLE_MODELS}/${FILE_WE}_done.txt)
+        set(OUT_FILES ${OUT_DONE_FILE} ${OUT_FILES})
+        add_custom_command(OUTPUT ${OUT_DONE_FILE}
+                COMMAND ${PYTHON_EXECUTABLE}
+                ${CMAKE_CURRENT_SOURCE_DIR}/test_models/gen_wrapper.py
+                ${GEN_SCRIPT}
+                ${TEST_PADDLE_MODELS}
+                ${OUT_DONE_FILE}
+                DEPENDS ${PADDLE_ALL_SCRIPTS}
+                )
+    endforeach()
+    add_custom_target(paddlepaddle_test_models DEPENDS ${OUT_FILES})
+
+    install(DIRECTORY ${TEST_PADDLE_MODELS}
```

Comment:
Should we move this code to common `cmake/test_model_zoo.cmake`? It will be able to convert / copy all models (ONNX, PDPD if we go this way)

In ideal case, I want to have a common place for all models since they are shared between IE / ngraph and used in both places. So, such test model zoo will be common and have a dedicated location in source tree as well

Reply:
I thought trend was opposite - have frontend-specific stuff in frontend-specific place. E.g. everything that is related to 'paddlepaddle' should be stored in frontend/paddlepaddle folder (maybe even tests should go in future from `ngraph/tests/frontend/paddlepaddle` to `ngraph/frontend/paddlepaddle/tests`
If this is correct we should possibly even avoid having pdpd-specific tests inside inference-engine/tests and move such tests to 'pdpd-folder' with dependency on IE